workers/task-driver/src/tasks/node_startup.rs

```diff
@@ -234,16 +308,100 @@ impl NodeStartupTask {
 
         info!("initializing raft with {} peers", peers.len());
         self.state.initialize_raft(peers).await?;
+
         Ok(())
     }
 
+    /// Setup the relayer's wallet, the wallet at which the relayer will receive
+    /// fees
+    ///
+    /// If the wallet is found on-chain, recover it. Otherwise, create a new one
+    async fn setup_relayer_wallet(&self) -> Result<(), NodeStartupTaskError> {
+        let chain_id = self
+            .arbitrum_client
+            .chain_id()
+            .await
+            .map_err(err_str!(NodeStartupTaskError::Arbitrum))?;
+
+        // Derive the keychain, blinder seed, and share seed from the relayer pkey
+        let blinder_seed = derive_blinder_seed(&self.keypair)
+            .map_err(err_str!(NodeStartupTaskError::DeriveWallet))?;
+        let share_seed = derive_share_seed(&self.keypair)
+            .map_err(err_str!(NodeStartupTaskError::DeriveWallet))?;
+        let keychain = derive_wallet_keychain(&self.keypair, chain_id)
+            .map_err(err_str!(NodeStartupTaskError::DeriveWallet))?;
+
+        // Set the node metadata entry for the relayer's wallet
+        let wallet_id = derive_wallet_id(&self.keypair)
+            .map_err(err_str!(NodeStartupTaskError::DeriveWallet))?;
+
+        // Attempt to find the wallet on-chain
+        if self.find_wallet_onchain(wallet_id, blinder_seed, share_seed, keychain.clone()).await? {
+            info!("found relayer wallet on-chain");
+            return Ok(());
+        }
+
+        // Otherwise, create a new wallet
+        self.create_wallet(wallet_id, blinder_seed, share_seed, keychain).await
+    }
+
     /// Manage the process to join an existing raft cluster
     ///
     /// TODO: Implement snapshot & promotion requests if joining an existing
     /// cluster
     #[allow(clippy::unused_async)]
     async fn join_raft(&self) -> Result<(), NodeStartupTaskError> {
-        println!("joining raft cluster");
+        info!("joining raft cluster");
         Ok(())
     }
+
+    // -----------
+    // | Helpers |
+    // -----------
+
+    /// Attempt to fetch a wallet from on-chain
+    async fn find_wallet_onchain(
+        &self,
+        wallet_id: WalletIdentifier,
+        blinder_seed: Scalar,
+        share_seed: Scalar,
+        keychain: KeyChain,
+    ) -> Result<bool, NodeStartupTaskError> {
+        info!("Finding relayer wallet on-chain");
+        let descriptor =
+            LookupWalletTaskDescriptor::new(wallet_id, blinder_seed, share_seed, keychain)
```

Comment:
do we need this lookup task to bypass the queue as well? probably not, since the raft should already be initialized by this point, right?

Reply:
correct, once we're initialized the raft we're good to go with task queue. it's only the original task that sets up the raft which requires we bypass